File: screen-switcher/src/main/java/com/jaynewstrom/screenswitcher/ScreenPopListener.java
package com.jaynewstrom.screenswitcher;

/**
 * It's important that this is only implemented on objects that live outside of the {@link android.app.Activity} lifecycle.
 */
public interface ScreenPopListener {

    /**
     * @param screen The {@link Screen} that is trying to be popped.
     * @return true if the pop has been consumed and the {@link Screen} should not be popped.
     */
    boolean onScreenPop(Screen screen);
}

File: screen-switcher/src/main/java/com/jaynewstrom/screenswitcher/ScreenPopListener.kt
package com.jaynewstrom.screenswitcher

import android.view.View

/**
 * It's important that this is only implemented on objects that live outside of the [android.app.Activity] lifecycle.
 */
interface ScreenPopListener {
    /**
     * @param view The [View] associated with the screen that is trying to be popped.
     * @param popContext An opaque object that is passed from the pop source to be used by pop listeners.
     *
     * @return true if the pop has been consumed and the [Screen] should not be popped.
     */
    fun onScreenPop(view: View, popContext: Any?): Boolean
}
